### PR TITLE
fix(eslint-plugin): fix no-base-to-string boolean literal check

### LINKED ISSUE #1655: [@typescript-eslint/no-base-to-string] is complaining about boolean values
Are you opening an issue because the rule you're trying to use is not found?
No

**Repro**

Rule definition in YAML
```YAML
@typescript-eslint/no-base-to-string": error
```

Blamed TypeScript code fragment:
```TS
const result = compareSync(password, user.password); // result is of type boolean
logger.error(`AuthService.checkPassword(): ${result}`);
```

**Expected Result**
No output

**Actual Result**
`152:51  error  'result will evaluate to '[Object object]' when stringified  @typescript-eslint/no-base-to-string`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.21.1-alpha.5` |
| `@typescript-eslint/parser`        | `2.21.1-alpha.5` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `7.0.0-alpha.2` |
| `node`                             | `13.9.0` |
| `npm`                              | `6.14.1` |

### PR DESCRIPTION
**Pull request by duduluu**
Fixes #1655
Feat: add option ignoreTypeNames

### CODE REVIEW

**Comment by astorije** [comment]
@duduluu, thanks for this! With regards to @felixfbecker's comment at https://github.com/typescript-eslint/typescript-eslint/issues/1655#issuecomment-593659511, do you want to support other types here?

Selfishly I'm hitting this issue with `RegExp`s, so that would be helpful :)

```ts
> /myRegExp/.toString()
'/myRegExp/'
```

**Comment by bradzacher** [comment]
just `RegExp` will do, the other two should already pass the checks as they have `toString`

**Comment by duduluu** [comment]
Sorry for late, let me try.

**Comment by duduluu** [comment]
Hi, @bradzacher . Does that mean this PR is unnecessary after the TypeScript merges your fixes for base types' definition?

**Comment by felixfbecker** [comment]
No, because there can be people on older TypeScript versions :)

File: packages/eslint-plugin/src/rules/no-base-to-string.ts [hunk 1]
```diff
@@ -42,16 +43,23 @@ export default util.createRule<Options, MessageIds>({
...
+              type: 'string',
+            },
+          },
         },
         additionalProperties: false,
       },
     ],
     type: 'suggestion',
   },
   defaultOptions: [{ ignoreTaggedTemplateExpressions: true }],
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/no-base-to-string.ts:58]: I think it's 100% okay to default it to this, as that's what people would expect

```suggestion
  defaultOptions: [
    {
      ignoreTaggedTemplateExpressions: true,
      ignoredTypeNames: ['RegExp'],
    },
  ],
```

Comment (duduluu) [reply packages/eslint-plugin/src/rules/no-base-to-string.ts]: done :)

**Pull request merged**